### ISSUE TITLE
Additional Arguments Support

### ISSUE THREAD

**Issue by mia-n** [podusowski/rhorizons#15]
Hello! I don't know if this is actively maintained, but I was playing around with it and I noticed that you're [hardcoding the CENTER param to the sun's body center](https://github.com/podusowski/rhorizons/blob/92c039e8e46db980ac90fc8a4151ab5e54e05efa/src/client.rs#L112-L115).

This gives pretty nonsensical results if examining natural satellites. For instance, for ephemeris of saturn's moon Titan you'd really want the CENTER to be either 500@699 (the body center of Saturn itself) or 500@6 (the Saturn System Barycenter).

For my purposes I forked your repo and just added an extra arg (centerID) to the client function instead of passing it in hardcoded.

Then I started playing around and realized there were even more arguments I'd like to be able to tweak - for instance, passing in STEP_SIZE in addition to START_TIME and STOP_TIME, or passing in the TLIST parameter instead of START_TIME and STOP_TIME.

Adjusting REF_PLANE is also super useful for getting correct-looking output for natural satellites.

I'm happy to make a pull request with some adjustments to this effect, but I thought I'd write this up to get your input first.

There's a few ways one could go about it and I don't have a strong opinion on style. Some options:
1. Update ephemeris_orbital_elements and ephemeris_vector to accept many more arguments in their function signature
2. Update them to accept a struct of Optional<T> ephemeris-specific parameters (and then serialize that into the query-string tuple vec, leaving out any that aren't specified). I think this is a little cleaner and more flexible, but it does require potential clients to build the request struct.
3. Both of the above break the existing API contract, which if you want to avoid I could just make totally new functions that act like one of the above options but leaving the existing API contract intact.
4. Something else?

**Comment by NiklasVousten** [comment]
I am currently working on a Command Builder, which allows using these additional arguments. My last update was a while ago, but I plan to continue working on it soon.

See here: [CommandBuilder Branch](https://github.com/NiklasVousten/rhorizons/tree/CommandBuilder)

**Comment by mia-n** [comment]
Oh, awesome! Cool :D

**Comment by mvernacc** [comment]
@g1aeder I saw your CommandBuilder branch and it looks useful for a project I'm working on. Do you expect to finish it?

**Comment by podusowski** [comment]
Would be happy to accept it once ready. :) Sorry for not paying too much attention to this crate.

**Comment by NiklasVousten** [comment]
Yes I am still working on it. I pushed the current progress to weeks ago, with an API change (Old one available with a linked commit). Not sure how the interfacing should be done. The current idea is to use structs and enums to create the command, which is additionally validated. 
Not sure if this is the best approach.
Validation would be also done by the Horizon-API, so this might not be needed.

Happy for input from your sides
